eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/BundleBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.core.Box;
+import com.schedjoules.eventdiscovery.framework.serialization.core.FluentBuilder;
+import com.schedjoules.eventdiscovery.framework.serialization.core.Key;
+
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.Present;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class BundleBuilder implements FluentBuilder<Bundle>
+{
+    private final Bundle mBundle;
+
+
+    public BundleBuilder(Bundle bundle)
+    {
+        mBundle = bundle;
+    }
+
+
+    public BundleBuilder()
+    {
+        this(new Bundle());
+    }
+
+
+    @Override
+    public Bundle build()
+    {
+        return mBundle;
+    }
+
+
+    @Override
+    public <T> FluentBuilder<Bundle> withParam(Key<T> key, Box<T> box)
+    {
+        return withParam(key, new Present<>(box));
+    }
+
+
+    @Override
+    public <T> FluentBuilder<Bundle> withParam(Key<T> key, Optional<Box<T>> optBox)
+    {
+        Bundle bundle = new Bundle(mBundle);
```

Comment:
I was not sure about this, what would be the reasonable contract to this kind of immutable object 'with' method with optional parameter, should it return new instance or not.
But in practice, I see that we will probably not use the same builder to create multiple Bundles or Intents, so I changed it. I will also add note to the javadoc.